misc/scripts/copyright_headers.py

```diff
@@ -73,7 +73,7 @@
         line = fileread.readline()
         header_done = False
 
-        while line.strip() == "":  # Skip empty lines at the top
+        while line.strip() == "" and line != "":  # Skip empty lines at the top
```

Comment:
I think the recommended way to parse files in Python is
```Python
while fileread:
```
not
```Python
while fileread.readline() != "":
```

I'd recommend changing the code to something like
```Python
while fileread:
    line = fileread.readline()
    if line == "":
        continue
```

Reply:
This is not how file works in python the `fileread` variable will not be considered as false when end of file is reached. This means `while fileread:` will be an infinite loop. Moreover here we want to remove only the empty lines at the start of the file which is exactly what it was doing. Code was ok just didn't think of empty file.

Reply:
Actually, this should probably be rewritten to use for loops instead of while loops. It looks like `for line in file` is the recommended way to iterate through a file line by line.

Reply:
In fact I agree that for loop are more idiomatic in file reading in pythonbut both can be found, moreover here changing the while loop wouldn't be recommended because we are first skipping lines ***until*** we find a non empty line then skipping ***until*** we find a non header line. Those ***until*** are for me a sign that while loops are more appropriate. Moreover I don't think a refactor of the file would be a good idea when the fix is simply modifying one line and therefore has no link with the refactor.

Reply:
Fair enough. I tend to get scared of `while` loops that don't have some guaranteed termination condition, but it's hard to argue with the simplicity of a one line fix.

Reply:
the `line != ""` is in fact the guaranteed termination because we are sure it will evaluate to false at some point because the file as a finite number of lines and each loop will consume one line hence it will either terminate the loop because of the other condition or reach the end of the file and terminate the loop because of this condition.